### ISSUE TITLE
Template Subfolder convention in README/existing tempaltes

### ISSUE THREAD

**Issue by kevya-google** [abcxyz/abc#166]
### TL;DR

Establish a recommended convention of having template files that are intended to be included placed in a subfolder one directory below the spec.yaml. This way, other files that are repo-specific (such as .vscode and .gitignore) or metadata for the template won't need to be excluded as part of the spec.yaml (for example, imagine a README.md for describing how to use the template on the same level as the spec.yaml and another README.md in the template folder meant to be included in the render describing details about the template code). However, this shouldn't be strictly enforced, providing some flexibility to template authors as there could be some templates that have a need for follow up actions, different files for different environments, other layout requirements, etc.

Context: https://docs.google.com/document/d/12iN2g1Idz6fW5ZqaokTmSFZf5kHXaWeewULWy2q5GfE/edit#heading=h.z1xyeclrumgj
